workers/tests/integrationtests/test_oqt.py

```diff
@@ -14,157 +20,114 @@
         )
-        self.assertIsNotNone(indicator.result.label)
-        self.assertIsNotNone(indicator.result.value)
-        self.assertIsNotNone(indicator.result.description)
-        self.assertIsNotNone(indicator.result.svg)
+        indicator = asyncio.run(oqt.create_indicator(param))
+        self.run_tests(indicator)
 
     @oqt_vcr.use_cassette()
     def test_create_indicator_dataset_custom_fid_field_int(self):
-        indicator = asyncio.run(
-            oqt.create_indicator(
-                "GhsPopComparisonBuildings",
-                "building_count",
-                dataset=self.dataset,
-                feature_id=self.feature_id,
-                fid_field=self.fid_field,
-            )
+        param = IndicatorDatabase(
+            name=self.name,
+            layerName=self.layer_name,
+            dataset=self.dataset,
+            featureId=self.feature_id,
+            fidField=self.fid_field,
         )
-        self.assertIsNotNone(indicator.result.label)
-        self.assertIsNotNone(indicator.result.value)
-        self.assertIsNotNone(indicator.result.description)
-        self.assertIsNotNone(indicator.result.svg)
+        indicator = asyncio.run(oqt.create_indicator(param))
+        self.run_tests(indicator)
 
     @oqt_vcr.use_cassette()
     def test_create_indicator_dataset_custom_fid_field_str(self):
-        indicator = asyncio.run(
-            oqt.create_indicator(
-                "GhsPopComparisonBuildings",
-                "building_count",
-                dataset=self.dataset,
-                feature_id="Heidelberg",
-                fid_field="name",
-            )
+        param = IndicatorDatabase(
+            name=self.name,
+            layerName=self.layer_name,
+            dataset=self.dataset,
+            featureId="Heidelberg",
+            fidField="name",
         )
-        self.assertIsNotNone(indicator.result.label)
-        self.assertIsNotNone(indicator.result.value)
-        self.assertIsNotNone(indicator.result.description)
-        self.assertIsNotNone(indicator.result.svg)
+        indicator = asyncio.run(oqt.create_indicator(param))
+        self.run_tests(indicator)
 
     @oqt_vcr.use_cassette()
-    def test_create_indicator_from_database_invalid_arguments(self):
-        indicator = asyncio.run(
-            oqt.create_indicator(
-                "GhsPopComparisonBuildings",
-                "building_count",
-                dataset=self.dataset,
-                feature_id=self.feature_id,
-                feature=self.feature,
-            )
+    def test_create_report_bpolys(self):
+        """Test creating indicator from scratch."""
+        param = ReportBpolys(
+            name=self.report_name,
+            bpolys=self.feature,
         )
-        self.assertIsNotNone(indicator.result.label)
-        self.assertIsNotNone(indicator.result.value)
-        self.assertIsNotNone(indicator.result.description)
-        self.assertIsNotNone(indicator.result.svg)
```

Comment:
I'm not sure what this test was supposed to do, I think you deleted it on purpose, correct?

Reply:
It has been deleted on purpose yes. Not sure about the reason of this test either. The implementation does not what the title suggests.